edge-modules/DirectMethodCloudSender/src/Program.cs

```diff
@@ -123,14 +114,21 @@ static async Task<int> MainAsync()
                 {
                     await serviceClient.CloseAsync();
                 }
-
-                if (moduleClient != null)
-                {
-                    await moduleClient.CloseAsync();
-                }
             }
 
             Logger.LogInformation("CallDirectMethodFromCloud finished.");
         }
+
+        static void CallAnalyzerToReportStatus(string moduleId, CloudToDeviceMethodResult result, AnalyzerClient analyzerClient)
+        {
+            try
+            {
+                analyzerClient.AddDirectMethodStatusAsync(new ResponseStatus { ModuleId = moduleId, StatusCode = result.Status.ToString(), EnqueuedDateTime = DateTime.UtcNow });
+            }
+            catch (Exception e)
+            {
+                Logger.LogError(e.ToString());
```

Comment:
There should have another LogError method to provide a message and an exception.
```suggestion
                Logger.LogError(e, "Error when report result to Analyzer");
```